Formula/gitleaks.rb

```diff
@@ -15,7 +15,6 @@ class Gitleaks < Formula
   end
 
   depends_on "go" => :build
-  uses_from_macos "git", since: :big_sur # git 2.27+
```

Comment:
I think so, gitleaks needs `git` installed

Reply:
I think this can work with the system-installed Git, that's why I removed it.

Reply:
> I think this can work with the system-installed Git, that's why I removed it. If this runs on at least Git 2.14.3 (which is Homebrew's minimum required) then we shouldn't need it.

Sounds reasonable to me

Reply:
I'm not sure we should remove dependencies for that. @Homebrew/linux how would you normally do that?

Reply:
@SMillerDev?

Reply:
We only add `uses_from_macos "git"` if it needs a new version of Git. Otherwise users can use the same `git` that `brew` uses.